src/main/java/bio/terra/service/dataset/flight/ingest/IngestPopulateFileStateFromFlightMapStep.java

```diff
@@ -1,35 +1,72 @@
 package bio.terra.service.dataset.flight.ingest;
 
-import bio.terra.model.BulkLoadFileModel;
+import bio.terra.model.IngestRequestModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.exception.IngestFailureException;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
 import bio.terra.service.load.LoadService;
 import bio.terra.service.load.flight.LoadMapKeys;
 import bio.terra.stairway.FlightContext;
 import bio.terra.stairway.FlightMap;
 import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import java.util.ArrayList;
 import java.util.List;
-import java.util.Set;
 import java.util.UUID;
 import java.util.function.Predicate;
 
-// Populate the files to be loaded from the incoming array
 public class IngestPopulateFileStateFromFlightMapStep extends SkippableStep {
 
   private final LoadService loadService;
+  private final GcsPdao gcsPdao;
+  private final ObjectMapper objectMapper;
+  private final Dataset dataset;
+  private final int batchSize;
 
   public IngestPopulateFileStateFromFlightMapStep(
-      LoadService loadService, Predicate<FlightContext> skipCondition) {
+      LoadService loadService,
+      GcsPdao gcsPdao,
+      ObjectMapper objectMapper,
+      Dataset dataset,
+      int batchSize,
+      Predicate<FlightContext> skipCondition) {
     super(skipCondition);
     this.loadService = loadService;
+    this.gcsPdao = gcsPdao;
+    this.objectMapper = objectMapper;
+    this.dataset = dataset;
+    this.batchSize = batchSize;
   }
 
+  // For some reason, Spotbugs thinks the try-with-resources results in a redundant nullcheck...
+  // This appears to be a bug in Spotbugs. https://github.com/spotbugs/spotbugs/issues/756
   @Override
+  @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE")
```

Comment:
Instead of a comment you can use `justification` in the annotation. Here's the form that the stairway code uses, which summarizes the findbugs issue well.
```suggestion
  @Override
  @SuppressFBWarnings(
    value = "RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE",
    justification = "Spotbugs doesn't understand resource try construct")
```